### PR TITLE
Update Forge2D docs

### PR DESCRIPTION
**Pull request by spydon**
# Description

Update the Forge2D docs to reflect the newest changes

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `v1.0.0`
- [x] This PR is targeted to merge into `v1.0.0` (not `master` or `develop`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: doc/forge2d.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Forge2D
+
+We (the Flame organization) maintains a ported version of the Box2D physics engine and our version is called Forge2D.
```
Comment (erickzanardo) [inline doc/forge2d.md:3]: I am not 100% sure, but I think here should `maintain`

File: doc/forge2d.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+The Forge2D project can use can be found [here](https://github.com/flame-engine/forge2d), and it can be used in any Dart project.
```
Comment (erickzanardo) [inline doc/forge2d.md:5]: this phrase is a little confusing, `can use can be found`

File: doc/forge2d.md (hunk 1 above)
Comment (spydon) [reply doc/forge2d.md:3]: You can say both :)

File: doc/forge2d.md (hunk 2 above)
Comment (spydon) [reply doc/forge2d.md:5]: Good catch!

Comment (spydon) [reply doc/forge2d.md]: Fixed

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**